hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -955,3 +959,80 @@ def array_for(index_shape, size):
     return result_shape.flatmap(
         lambda index_shape: st.tuples(*(array_for(index_shape, size) for size in shape))
     )
+
+
+def make_noncontiguous(array: np.ndarray, stride: int = 2) -> np.ndarray:
+    """Return a noncontiguous view of `array` if possible."""
+    if np.isscalar(array) or array.size < 2:
+        return array
+
+    # Flatten our data, interleave it with garbage, then create a view that only
+    # looks at the original data.
+    flat = array.ravel()
+    backing_arr = np.empty(flat.size * abs(stride), dtype=flat.dtype)
+    backing_arr[::stride] = flat
+    return backing_arr[::stride].reshape(array.shape)
+
+
+def permute_dimensions(array: np.ndarray, permuted_indices: Sequence[int]) -> np.ndarray:
+    """Return an array that is functionally identical to `array`, but whose underlying dimensions have been permuted."""
+    assert len(array.shape) == len(permuted_indices)
+
+    # Invert the permutatation so that when we apply the original permutation, we get back
+    # our desired data.
+    inverse_permutation = [0] * len(permuted_indices)
+    for i, index in enumerate(permuted_indices):
+        inverse_permutation[index] = i
+
+    result = array.transpose(inverse_permutation)
+    result = np.ascontiguousarray(result)
+    result = result.transpose(permuted_indices)
+
+    return result
+
+
+class ArrayMemoryScramblerStrategy(st.SearchStrategy):
+    def __init__(self):
+        pass
+
+    def do_draw(self, data):
+        # 0 is an invalid stride. 1 is a no-op, but rather than draw a boolean to control whether
+        # we make it discontiguous, just use stride=1.
+        stride = data.draw(st.integers(-5, 5).filter(lambda x: x != 0))
```

Comment:
I wonder if we might make the max-stride configurable so that users can control the memory footprint of the scrambler. Thoughts?